libV2/utils/timestamp.js

```diff
@@ -42,8 +42,45 @@ function now() {
     return Date.now() * 1000;
 }
 
+/**
+ * Slice the time range represented by the passed timestamps
+ * into slices of at most `step` duration.
+ *
+ * Both `start` and `end` are included in the returned slices.
+ * Slice timestamps are inclusive and non overlapping.
+ *
+ * For example sliceTimeRange(0, 5, 2) will yield
+ * [0, 1]
+ * [2, 3]
+ * [4, 5]
+ *
+ * @param {Number} start
+ * @param {Number} end
+ * @param {Number} step
+ */
+
+function* sliceTimeRange(start, end, step) {
+    let spos = start;
+    let epos = start + step - 1;
+    while (true) {
```

Comment:
A minor improvement/clarification of this loop could be:

```javascript
while (epos < end) {
    yield [spos, epos];
    spos += step;
    epos += step;
}
yield [spos, end];
```

Reply:
I actually had this to start with and changed it because I though there was an edge case it didn't handle. Upon a second look that edge case isn't possible :D